### PR TITLE
Add release workflows

### PR DESCRIPTION
**Pull request by gaiksaya**
### Description
This PR adds below things:
* A GitHub action workflow that will be triggered when a tag is pushed to this repository. The workflow builds the product and creates a draft release with built artifacts attached as artifacts.tar.gz in the release.
* A jenkins workflow that is triggered based on above draft release. The workflow looks for `artifacts.tar.gz`, downloads it and then signs it and publishes it to [PyPi](https://pypi.org/) 

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gaiksaya** [comment]
Getting few things in place so please wait before merging. Please feel free to review till then.
Thanks!

**Comment by VachaShah** [comment]
Workflows are failing for certain python versions for the latest ubuntu-22.04. Related issue on `actions/setup-python` https://github.com/actions/setup-python/issues/543

**Review by dblock — DISMISSED** [review]

**Review by dblock — DISMISSED** [review]

File: RELEASING.md [hunk 1]
```diff
@@ -31,4 +31,10 @@ Repositories create consistent release labels, such as `v1.0.0`, `v1.1.0` and `v
...
 
-The release process is standard across repositories in this org and is run by a release manager volunteering from amongst [MAINTAINERS](MAINTAINERS.md).
\ No newline at end of file
+The release process is standard across repositories in this org and is run by a release manager volunteering from amongst [maintainers](MAINTAINERS.md).
+
+1. Create a tag, e.g. v2.1.0, and push it to the GitHub repo.
+1. The [release-drafter.yml](.github/workflows/release-drafter.yml) will be automatically kicked off and a draft release will be created.
+1. This draft release triggers the [jenkins release workflow](https://build.ci.opensearch.org/job/opensearch-py-release/) as a result of which opensearch-py client is released on [PyPi](https://pypi.org/project/opensearch-py/).
+1. Once the above release workflow is successful, the drafted release on GitHub is published automatically.
+1. Increment "version" in [_version.py](./opensearchpy/_version.py) to the next patch release, e.g. v2.1.1. See [example](https://github.com/opensearch-project/opensearch-py/pull/167)
```
Comment (dblock) [inline RELEASING.md:40]: Needs a period ;)

Comment (gaiksaya) [reply RELEASING.md]: Added :)

**Comment by gaiksaya** [comment]
Everything is set up. Once this PR is merged, I'll be creating the webhooks that will trigger the release when a tag is pushed.
Please review.
Thanks!

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
@VachaShah Good with you?

**Comment by gaiksaya** [comment]
> @VachaShah Good with you?

I believe she is out till end of next month. Do you still want to wait till then or get reviewed by anyone else?

**Review by harshavamsi — APPROVED** [review]

**Comment by gaiksaya** [comment]
Webhook is added now! This repo is set for automated release once a tag is pushed. Please engage @opensearch-project/engineering-effectiveness in case of any failures or help.
Thanks!

**Pull request merged**